hal/armv7m/arch/elf.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * Hardware Abstraction Layer ELF support
+ *
+ * Copyright 2024 Phoenix Systems
+ * Author: Hubert Badocha
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#define R_ARM_ABS32   2
+#define R_ARM_TARGET1 38
+
+
+static inline int hal_isRelReloc(int relType)
+{
+	return (relType == R_ARM_ABS32) || (relType == R_ARM_TARGET1);
```

Comment:
MISRA

Side note, I'm not sure if this approach is scalable. Might be actually more useful to be able to just know relocation type in the process_load.
```suggestion
	return ((relType == R_ARM_ABS32) || (relType == R_ARM_TARGET1)) ? 1 : 0;
```

Reply:
When it comes to the scalability its a hard call without having more relocations. To support FDPIC format I will have to extend the list in this function on ARM, and add one other type checking function. That sound ok to me